DotNetWebApi/Controllers/PicnicController.cs

```diff
@@ -0,0 +1,118 @@
+                            Location = picnic.Location,
+                            StartTime = picnic.StartTime,
+                            HasMusic = picnic.HasMusic == true,
+                            HasFood = picnic.HasFood == true,
+                            TeddyBears = picnic.TeddyBears.Select(tb => tb.Name)
+                        };
+        return await picnics.ToListAsync();
+    }
+
+    [HttpGet("{name}")]
+    public async Task<ActionResult<PicnicReturn>> GetPicnicByName(string name)
+    {
+        var picnics = from picnic in _context.Picnics
+                        where picnic.PicnicName == name
+                        select new PicnicReturn () {
+                            Id = picnic.Id,
+                            PicnicName = picnic.PicnicName,
+                            Location = picnic.Location,
+                            StartTime = picnic.StartTime,
+                            HasMusic = picnic.HasMusic == true,
+                            HasFood = picnic.HasFood == true,
+                            TeddyBears = picnic.TeddyBears.Select(tb => tb.Name)
+                        };
+        var picnicFound = await picnics.FirstOrDefaultAsync();
+        if (picnicFound == null){
+            return NotFound();
+        }
+        //otherwise
+        return picnicFound;
+    }
+
+    [HttpPost("")]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    public async Task<ActionResult<PicnicCreate>> CreatePicnic(PicnicCreate picnic)
+    {
+        var location = await _context.PicnicLocations.Where(p => p.LocationName == picnic.LocationName).FirstOrDefaultAsync();
+        var teddyBears = await _context.TeddyBears.Where(t => picnic.TeddyBears.Contains(t.Name)).ToListAsync();
+        var newPicnic = new Picnic
+        {
+            PicnicName = picnic.PicnicName,
+            Location = location,
+            StartTime = picnic.StartTime,
+            HasFood = picnic.HasFood,
+            HasMusic = picnic.HasMusic,
+        };
+        // add in the teddy bears
+        foreach (var bear in teddyBears){
+            newPicnic.TeddyBears.Add(bear);
+        }
+        _context.Picnics.Add(newPicnic);
+        await _context.SaveChangesAsync();
+
+        //We can't just return the new picnic, we'll get circular JSON issues, so...
+        var picnicToReturn = new PicnicReturn
+        {
+            Id = newPicnic.Id,
+            PicnicName = newPicnic.PicnicName,
+            Location = newPicnic.Location,
+            StartTime = newPicnic.StartTime,
+            HasFood = newPicnic.HasFood ?? false,
+            HasMusic = newPicnic.HasMusic ?? false,
+            TeddyBears = newPicnic.TeddyBears.Select(t => t.Name),
+        };
+        picnicToReturn.Location?.Picnics?.Clear();     //more cleanup for circular JSON issues
+
+        return CreatedAtAction(nameof(CreatePicnic), new { id = newPicnic.Id }, picnicToReturn);
+    }
+
+    [HttpGet("{locationName}")]
```

Comment:
I think this route conflicts with `GetPicnicByName`.  I think it was prefixed with "Locations/" in the tutorial.